packages/high-roller-bot/src/utils.ts

```diff
@@ -78,42 +86,30 @@ export async function deposit(
       } as NodeTypes.DepositParams
     });
 
-    const updatedFreeBalance = await getFreeBalance(node, multisigAddress);
-    const updatedAliceBalance = updatedFreeBalance.state.aliceBalance;
-    const updatedBobBalance = updatedFreeBalance.state.bobBalance;
-
-    if (updatedAliceBalance.gt(aliceBalance)) {
-      console.info("Waiting for counter party to deposit same amount");
-      while (
-        !(await getFreeBalance(node, multisigAddress)).state.bobBalance.gt(
-          updatedBobBalance
-        )
-      ) {
-        console.info(
-          `Waiting ${DELAY_SECONDS} more seconds for counter party deposit`
-        );
-        await delay(DELAY_SECONDS * 1000);
-      }
-    } else if (updatedFreeBalance.state.bobBalance.gt(bobBalance)) {
-      console.info("Waiting for counter party to deposit same amount");
-      while (
-        !(await getFreeBalance(node, multisigAddress)).state.aliceBalance.gt(
-          updatedAliceBalance
-        )
-      ) {
-        console.info(
-          `Waiting ${DELAY_SECONDS} more seconds for counter party deposit`
-        );
-        await delay(DELAY_SECONDS * 1000);
-      }
-    } else {
-      throw Error(`Neither balance was updated.\n
-        Original Free Balance: alice: ${aliceBalance}, bob: ${bobBalance}\n
-        Updated Free Balance: alice: ${updatedAliceBalance}, bob: ${updatedBobBalance}
-      `);
+    const postDepositBalances = await getFreeBalance(node, multisigAddress);
+
+    if (
+      !postDepositBalances[myFreeBalanceAddress].gt(
+        preDepositBalances[myFreeBalanceAddress]
+      )
+    ) {
+      throw Error("My balance was not updated.");
+    }
+
+    console.info("Waiting for counter party to deposit same amount");
+
+    while (
+      !(await getFreeBalance(node, multisigAddress))[
+        counterpartyFreeBalanceAddress
+      ].gt(preDepositBalances[counterpartyFreeBalanceAddress])
+    ) {
```

Comment:
Wrap this 3 line conditional in a named function e.g., `checkFreeBalanceGreaterThan(...)`